library/std/src/sys/windows/path/absolute/tests.rs

```diff
@@ -0,0 +1,149 @@
+// Original implementation taken from crate "normpath".
+// Copyright 2020 "dylni" (https://github.com/dylni)
+
+// Most test cases were taken from:
+// https://googleprojectzero.blogspot.com/2016/02/the-definitive-guide-on-win32-to-nt.html
+
+// These tests assume that the current drive is "C:".
+
+use crate::env;
+use crate::io;
+use crate::path::Path;
+
+fn assert_eq(left: &Path, right: &Path) {
+    assert_eq!(left.as_os_str().bytes(), right.as_os_str().bytes());
+}
+
+fn test(expected: &str, path: &str) {
+    assert_eq(Path::new(expected), &super::absolute(Path::new(path)).unwrap());
+}
+
+macro_rules! test_unchanged {
+    ( $path:expr ) => {
+        test($path, $path);
+    };
+}
+
+#[test]
+fn test_drive_absolute() {
+    test_unchanged!(r"C:\ABC\DEF");
+    test_unchanged!(r"C:\");
+    test_unchanged!(r"C:\ABC\");
+    test(r"C:\ABC\DEF", r"C:\ABC\DEF. .");
+    test(r"C:\ABC\DEF", r"C:/ABC/DEF");
+    test(r"C:\XYZ", r"C:\ABC\..\XYZ");
+    test(r"C:\", r"C:\ABC\..\..\..");
+}
+
+#[test]
+fn test_drive_relative() {
+    test(r"X:\DEF\GHI", r"X:DEF\GHI");
+    test(r"X:\", r"X:");
+    test(r"X:\DEF", r"X:DEF. .");
+    test(r"Y:\", r"Y:");
+    test(r"Z:\", r"Z:");
+    test(r"X:\XYZ", r"X:ABC\..\XYZ");
+    test(r"X:\", r"X:ABC\..\..\..");
+}
+
+#[test]
+fn test_rooted() {
+    test(r"C:\ABC\DEF", r"\ABC\DEF");
+    test(r"C:\", r"\");
+    test(r"C:\ABC\DEF", r"\ABC\DEF. .");
+    test(r"C:\ABC\DEF", r"/ABC/DEF");
+    test(r"C:\XYZ", r"\ABC\..\XYZ");
+    test(r"C:\", r"\ABC\..\..\..");
```

Comment:
These tests assume the current directory is on the `C:` drive. That might not always be true.

Reply:
Yes, that's mentioned here:
https://github.com/rust-lang/rust/pull/94887/files/7d1be5cc3ee6269c8201650de709bd1be9517d51#diff-07b40c9e9f3c35cf76a50ac526ab31e19ffe1617a48aed94a731770727c9982fR7

Unfortunately, for these tests to be useful, they need to make that assumption or become much more complicated.